### PR TITLE
allow spawn eggs to do SOMETHING

### PR DESCRIPTION
**Pull request by ayunami2000**
they now spawn a mob despite the mob having zero nbt from the spawn egg. "it's better than nothing" as one might say.

fixes (until we implement partial nbt support) https://totalfreedom.atlassian.net/browse/FS-377

### CODE REVIEW

File: src/main/java/me/totalfreedom/totalfreedommod/blocking/MobBlocker.java [hunk 1]
```diff
@@ -31,11 +31,13 @@ public void onStop()
     {
     }
 
+    //after careful review by ayunami2000 this code always returns before doing anything
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/blocking/MobBlocker.java:34]: @Paldiu Up to you to determine if we should make a new PR for this - ideally, each PR should only contain stuff related to the ticket. This change does not fit the scope of the current ticket.

**Review by Focusvity — CHANGES_REQUESTED** [review]
Please format the changes to Allman style

File: src/main/java/me/totalfreedom/totalfreedommod/blocking/MobBlocker.java (hunk 1 above)
Comment (ayunami2000) [reply src/main/java/me/totalfreedom/totalfreedommod/blocking/MobBlocker.java]: Ok, I removed my change to this with my latest commit, but I guess it needs to be put in a Jira ticket to be looked further into? Also I reformatted the rest of it to Allman style with my latest commit as well.

File: src/main/java/me/totalfreedom/totalfreedommod/blocking/InteractBlocker.java [hunk 2]
```diff
@@ -79,9 +81,28 @@ private void handleRightClick(PlayerInteractEvent event)
 
         if (Groups.SPAWN_EGGS.contains(event.getMaterial()))
         {
-            player.getInventory().clear(player.getInventory().getHeldItemSlot());
-            player.sendMessage(ChatColor.GRAY + "Spawn eggs are currently disabled.");
             event.setCancelled(true);
+            Block clickedBlock = event.getClickedBlock();
+            if (clickedBlock == null) {
```
Comment (Focusvity) [inline src/main/java/me/totalfreedom/totalfreedommod/blocking/InteractBlocker.java:86]: Bracket to be on newline

**Review by Focusvity — CHANGES_REQUESTED** [review]
Few more formatting to do

File: src/main/java/me/totalfreedom/totalfreedommod/blocking/InteractBlocker.java [hunk 3]
```diff
@@ -79,9 +81,23 @@ private void handleRightClick(PlayerInteractEvent event)
...
+            if (clickedBlock == null) return;
+            EntityType eggType = null;
+            try {
+                Material mat = event.getMaterial();
+                if (mat == Material.MOOSHROOM_SPAWN_EGG) {
+                    eggType = EntityType.MUSHROOM_COW;
+                } else {
+                    eggType = EntityType.valueOf(mat.name().substring(0, mat.name().length() - 10));
+                }
+            } catch (IllegalArgumentException ignored) {}
```
[the remark this answers is not in the document]
Comment (Focusvity) [reply src/main/java/me/totalfreedom/totalfreedommod/blocking/InteractBlocker.java:102]: @ayunami2000 Missing this formatting

[the remark this answers is not in the document]
Comment (ayunami2000) [reply src/main/java/me/totalfreedom/totalfreedommod/blocking/InteractBlocker.java]: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: :skull: done

**Comment by Focusvity** [comment]
I will let the checks complete before finalising my review.

**Review by Focusvity — DISMISSED** [review]
LGTM!

**Review by Paldiu — APPROVED** [review]

**Pull request merged**